src/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,222 @@
+        }
+      });
+    }
+  }
+
+  _initDatePicker = (type) => {
+    const date = (type === 'single')
+    ? this.element.querySelector(this.options.selectorDatePickerInput)
+    : this.element.querySelector(this.options.selectorDatePickerInputFrom);
+    const calendar = new Flatpickr(date, Object.assign(flattenOptions(this.options), {
+      allowInput: true,
+      mode: type,
+      onClose: (selectedDates) => {
+        this._updateInputFields(selectedDates, type);
+        if (type === 'range') {
+          if (calendar.selectedDates.length === 1) {
+            date.focus();
+          } else {
+            this.element.querySelector(this.options.selectorDatePickerInputTo).focus();
+          }
+        }
+      },
+      onChange: () => {
+        this._updateClassNames(calendar);
+        if (type === 'range') {
+          if (calendar.selectedDates.length === 1) {
+            this.element.querySelector(this.options.selectorDatePickerInputTo).classList.add('bx--focused');
+          } else {
+            this.element.querySelector(this.options.selectorDatePickerInputTo).classList.remove('bx--focused');
+          }
+        }
+      },
+      onMonthChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onYearChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onOpen: () => {
+        this._updateClassNames(calendar);
+      },
+      nextArrow: this._rightArrowHTML(),
+      prevArrow: this._leftArrowHTML(),
+    }));
+    if (type === 'range') {
+      this.element.querySelector(this.options.selectorDatePickerInputTo).addEventListener('click', () => {
+        this.element.querySelector(this.options.selectorDatePickerInputTo).focus();
+        calendar.open();
+      });
+      this._addInputLogic(this.element.querySelector(this.options.selectorDatePickerInputTo));
+    }
+    this.element.querySelector(this.options.selectorDatePickerIcon).addEventListener('click', () => {
+      calendar.open();
+    });
+    this._addInputLogic(date);
+    return calendar;
+  }
+
+  setMinDate = (date) => {
+    this.calendar.set('minDate', date);
+  }
+
+  setMaxDate = (date) => {
+    this.calendar.set('maxDate', date);
+  }
+
+  setDisabledDates = (dates) => {
+    this.calendar.set('disable', dates);
+  }
+
+  setDefaultDates = (dates) => {
+    this.calendar.set('defaultDate', dates);
+  }
+
+  _rightArrowHTML() {
+    return (`
+      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
+        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
+      </svg>`
+    );
+  }
+
+  _leftArrowHTML() {
+    return (`
+      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
+        <path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path>
+      </svg>`
+    );
+  }
+
+  _addInputLogic = (input) => {
+    const inputField = input;
+    inputField.addEventListener('change', () => {
+      if (inputField.value.length === 10) {
```

Comment:
Rather than checking the length which is format-specific, I’d suggest running `setDate()` *only* when `parseDate()` is successful.